authentication/src/main/java/com/shashank/splitterexpensemanager/authentication/login/LoginViewModel.kt

```diff
@@ -1,36 +1,85 @@
 package com.shashank.splitterexpensemanager.authentication.login
+
+import android.util.Log
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.shashank.splitterexpensemanager.core.network.NetworkCallState
 import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.database.DatabaseReference
+import com.google.firebase.database.FirebaseDatabase
+import com.shashank.splitterexpensemanager.authentication.login.repository.LoginRepository
+import com.shashank.splitterexpensemanager.authentication.model.Person
+import com.shashank.splitterexpensemanager.core.PERSON
+import com.shashank.splitterexpensemanager.core.PERSON_ID
+import com.shashank.splitterexpensemanager.core.SharedPref
+import com.shashank.splitterexpensemanager.localdb.model.Person as PersonEntity
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.tasks.await
 import javax.inject.Inject
+
 @HiltViewModel
-class LoginViewModel @Inject constructor() : ViewModel() {
+class LoginViewModel @Inject constructor(var loginRepository: LoginRepository) : ViewModel() {
+
+    @Inject
+    lateinit var sharedPref: SharedPref
     private var auth: FirebaseAuth = FirebaseAuth.getInstance()
+    private lateinit var databaseReference: DatabaseReference
```

Comment:
can we initialize databaseReference here ?

Reply:
done